### ISSUE TITLE
Benchmarks on ImageNet

### ISSUE THREAD

**Issue by mateuszmalinowski** [lim0606/lasagne-googlenet#1]
It would be useful if you benchmark your implementation on ImageNet, and provide the results.

**Comment by lim0606** [comment]
@mateuszmalinowski 

I couldn't train the model with lasagne because of computational time issue. 

It takes too much time in my system. (https://github.com/Lasagne/Lasagne/issues/312)

I shared the results trained with caffe (https://github.com/lim0606/caffe-googlenet-bn). 

The network architecture in this repository is different from the network architecture described in the batch normalization paper. (In here, I just add bn layer to all layers in original GoogLeNet)

However, the one trained with caffe has the same network architecture suggested in the paper.  

I will update if I train models with lasagne.

**Comment by Jiangfeng-Xiong** [comment]
> Some networks may have multiple output layers ... If you compute the network output expressions with separate calls to lasagne.layers.get_output(), they will not use the same samples. Furthermore, this may lead to unnecessary computation because Theano is not always able to merge identical computations properly.Calling get_output() only once prevents both of these issues.

the above comes from lasagne Documentation, i guess this might be be one reason why computational time issue in your implementation . @lim0606
